lib/src/iterable/iterable_extensions.dart

```diff
@@ -35,4 +37,72 @@ extension IterableExtensions<E> on Iterable<E> {
     }
     yield last;
   }
+
+  /// Zip the elements of the iterable together with those of the
+  /// given iterable to form a list of lists (of length two) of
+  /// elements of the same as those of the inputs..
+  ///
+  /// For example:
+  ///
+  /// [1, 2].zip([5, 6]) // [[1, 5], [2, 6]]
+  Iterable<List<E>> zip(Iterable<E> iterable) =>
+      zipTo(iterable, (left, right) => [left, right]);
+
+  /// Zip the elements of the iterable together with those of the
+  /// given iterable using the provided callback to combine each pair
+  /// of elements.
+  ///
+  /// If one iterable is shorter than the other, the extra elements
+  /// will be combined with `null`. Therefore, the callback
+  /// must be able to handle `null` for either of its parameters,
+  /// though not both.
+  ///
+  /// For example:
+  ///
+  /// ['Kate', 'Pablo'].zipToContainer(
+  ///   [28, 31],
+  ///   (name, age) => {name: age},
+  /// ); // [{'Kate': 28}, {'Pablo': 31}]
+  Iterable<C> zipTo<T, C>(
+    Iterable<T> iterable,
+    C Function(E, T) combine,
+  ) sync* {
+    final left = iterator;
```

Comment:
whoa, I'm definitely missing something here - what is `iterator`?

Reply:
`Iterable.iterator` is a getter that returns an iterator over the `Iterable`... https://api.dartlang.org/stable/2.7.0/dart-core/Iterator-class.html

I use it directly here so I can move through both `Iterable`s at the same time.

Reply:
oh cool, and since this is an extension method on `Iterable` you can use it directly. neat!

Reply:
Yeah, extension methods are A-MAZING!